src/Randomizer.App/Windows/GenerateRomWindow.xaml.cs

```diff
@@ -352,5 +501,81 @@ private void LogicCheckBox_Checked(object sender, RoutedEventArgs e)
             var property = type.GetProperty(checkBox.Name);
             property.SetValue(Options.LogicConfig, checkBox.IsChecked ?? false);
         }
+
+        /// <summary>
+        /// Updates to the dropdown to filter locations to specific regions
+        /// </summary>
+        /// <param name="sender">The dropdown that was updated</param>
+        /// <param name="e">The event object</param>
+        private void LocationsRegionFilter_SelectionChanged(object sender, SelectionChangedEventArgs e)
+        {
+            var comboBox = sender as ComboBox;
+            var selectedRegion = comboBox.SelectedItem as string;
+            foreach (FrameworkElement obj in LocationsGrid.Children) {
+                var location = obj.Tag as Location;
+                obj.Visibility = selectedRegion.Contains(location.Region.Name) ? Visibility.Visible : Visibility.Collapsed;
```

Comment:
This can cause a null reference exception if Tag is ever not a location or null.